sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -2011,6 +2011,9 @@ case class Deduplicate(
 }
 
 case class DeduplicateWithinWatermark(keys: Seq[Attribute], child: LogicalPlan) extends UnaryNode {
+  // Ensure that references include watermarking columns so they are not pruned away.
```

Comment:
nit: Watermark is a concept; it is more proper to call the column as "event time" column, as the column represents the event time of the input.

Title, and this line, and test name.